pygaggle/data/segmentation.py

```diff
@@ -32,17 +32,24 @@ def __init__(self, max_characters=10000000):
 
     def segment(self, documents: List[Text], seg_size: int, stride: int) -> SegmentGroup:
         segmented_doc, doc_end_indexes, end_idx = [], [0], 0
-
         for document in documents:
             doc = self.nlp(document.text[:self.max_characters])
             sentences = [sent.string.strip() for sent in doc.sents]
-            for i in range(0, len(sentences), stride):
-                segment_text = ' '.join(sentences[i:i + seg_size])
+            if len(sentences) == 0: # Text is empty
+                segment_text = document.title + '. '
                 segmented_doc.append(Text(segment_text, dict(docid=document.metadata["docid"])))
-                if i + seg_size >= len(sentences):
-                    end_idx += i/stride + 1
-                    doc_end_indexes.append(int(end_idx))
-                    break
+                end_idx += 1
+                doc_end_indexes.append(int(end_idx))
+            else:
+                for i in range(0, len(sentences), stride):
+                    segment_text = ' '.join(sentences[i:i + seg_size])
+                    if not document.title == '':
+                        segment_text = document.title + '. ' + segment_text
+                    segmented_doc.append(Text(segment_text, dict(docid=document.metadata["docid"])))
+                    if i + seg_size >= len(sentences):
+                        end_idx += i/stride + 1
+                        doc_end_indexes.append(int(end_idx))
+                        break
         return SegmentGroup(segmented_doc, doc_end_indexes)
 
     def aggregate(self, documents: List[Text], segments_group: SegmentGroup, method: str = "max") -> List[Text]:
```

Comment:
Add a short docstring to each method and one for the file describing what it does in general?

Reply:
Added one for the classes.  Is that good enough?

Reply:
It is good to have a docstring for each method, even if it is only one or two sentences.

Reply:
I added for the methods and classes.  I guess my question should have been whether I should add a docstring for the file?

Reply:
Sorry for the late reply. Ideally, there should be a docstring for the file, but I think it is fine docstrings for methods and classes only.